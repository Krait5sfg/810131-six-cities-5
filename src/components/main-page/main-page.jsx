import React from 'react';
import PropTypes from 'prop-types';
import PlaceCardList from '../place-card-list/place-card-list';
import {OfferPropTypes} from '../../utils/property-type';
import Map from '../map/map';
import {TypePage} from '../../utils/const';
import {connect} from 'react-redux';
import CityList from '../city-list/city-list';
import Sorting from '../sorting/sorting';
import NoPlace from '../no-place/no-place';
import YesPlace from '../yes-place/yes-place';

const MainPage = ({city, offers, onLinkEmailClick}) => {

  // опред классы для стр. в зависимости пустая или нет
  const isMainPageEmpty = offers.length ? false : true;
  const classNameForMainTag = isMainPageEmpty ? `page__main--index-empty` : ``;
  const classNameForPlaceContainer = isMainPageEmpty ? `cities__places-container--empty` : ``;

  return (
    <div className="page page--gray page--main">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <a className="header__logo-link header__logo-link--active">
                <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41" />
              </a>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <a className="header__nav-link header__nav-link--profile" href="#" onClick={onLinkEmailClick}>
                    <div className="header__avatar-wrapper user__avatar-wrapper">
                    </div>
                    <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>

      <main className={`page__main page__main--index ${classNameForMainTag}`}>
        <h1 className="visually-hidden">Cities</h1>
        <div className="tabs">
          <section className="locations container">
            <CityList />
          </section>
        </div>
        <div className="cities">
          <div className={`cities__places-container ${classNameForPlaceContainer} container`}>
            {isMainPageEmpty
              ? <NoPlace city={city} />
              : <YesPlace offerCount={offers.length} city={city}>
                <Sorting />
                <PlaceCardList offers={offers} typePage={TypePage.MAIN} />
              </YesPlace>
            }
            <div className="cities__right-section">
              {isMainPageEmpty
                ? ``
                : <Map
                  typePage={TypePage.MAIN}
                  city={city}
                  offers={offers}
                />}
            </div>
          </div >
        </div >
      </main >
    </div >
  );
};

MainPage.propTypes = {
  city: PropTypes.string.isRequired,
  onLinkEmailClick: PropTypes.func.isRequired,
  offers: PropTypes.arrayOf(OfferPropTypes).isRequired,
};

// связывает store c пропсами компонента
const mapStateToProps = (({city, offers}) => ({
  city,
  offers,
}));

export {MainPage};
export default connect(mapStateToProps)(MainPage);
