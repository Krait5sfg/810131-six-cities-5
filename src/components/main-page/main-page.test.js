import React from 'react';
import renderer from 'react-test-renderer';
import {MainPage} from './main-page';

const offers = [
  {
    city: ``,
    accommodation: {
      bedroomsCount: 0,
      features: [],
      guestsLimit: 0,
      isPremium: true,
      price: 0,
      rating: 0,
      title: ``,
      type: ``,
    },
    cityLocation: {latitude: 0, longitude: 0, zoom: 0},
    coordinates: [0, 0],
    description: ``,
    host: {
      avatar: ``,
      isPro: true,
      name: ``,
    },
    id: 1,
    images: [],
    isFavorite: false,
    previewImage: ``,
  },
  {
    city: ``,
    accommodation: {
      bedroomsCount: 0,
      features: [],
      guestsLimit: 0,
      isPremium: true,
      price: 0,
      rating: 0,
      title: ``,
      type: ``,
    },
    cityLocation: {latitude: 0, longitude: 0, zoom: 0},
    coordinates: [0, 0],
    description: ``,
    host: {
      avatar: ``,
      isPro: true,
      name: ``,
    },
    id: 2,
    images: [],
    isFavorite: false,
    previewImage: ``,
  },
  {
    city: ``,
    accommodation: {
      bedroomsCount: 0,
      features: [],
      guestsLimit: 0,
      isPremium: true,
      price: 0,
      rating: 0,
      title: ``,
      type: ``,
    },
    cityLocation: {latitude: 0, longitude: 0, zoom: 0},
    coordinates: [0, 0],
    description: ``,
    host: {
      avatar: ``,
      isPro: true,
      name: ``,
    },
    id: 3,
    images: [],
    isFavorite: true,
    previewImage: ``,
  },
];

jest.mock(`../header/header`, () => `Header`);
jest.mock(`../user/user`, () => `User`);
jest.mock(`../city-list/city-list`, () => `CityList`);
jest.mock(`../sorting/sorting`, () => `Sorting`);
jest.mock(`../place-card-list/place-card-list`, () => `PlaceCardList`);
jest.mock(`../map/map`, () => `Map`);

describe(`MainPage render correctly`, () => {
  it(`MainPage without offers`, () => {
    const tree = renderer
      .create(<MainPage
        city={`Amsterdam`}
        onLinkEmailClick={() => {}}
        offers={[]}
        onFavoriteButtonClick={() => {}}
        updateIdActiveCardForMap={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`MainPage with offers`, () => {
    const tree = renderer
      .create(<MainPage
        city={`Amsterdam`}
        onLinkEmailClick={() => {}}
        offers={offers}
        onFavoriteButtonClick={() => {}}
        updateIdActiveCardForMap={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
