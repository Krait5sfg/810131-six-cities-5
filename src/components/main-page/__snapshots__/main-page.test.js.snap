// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage render correctly MainPage with offers 1`] = `
<div
  className="page page--gray page--main"
>
  <Header>
    <User
      onLinkEmailClick={[Function]}
    />
  </Header>
  <main
    className="page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <CityList />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container  container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            3
             places to stay in 
            Amsterdam
          </b>
          <Sorting />
          <PlaceCardList
            offers={
              Array [
                Object {
                  "accommodation": Object {
                    "bedroomsCount": 0,
                    "features": Array [],
                    "guestsLimit": 0,
                    "isPremium": true,
                    "price": 241,
                    "rating": 4.4,
                    "title": "The Joshua Tree House",
                    "type": "room",
                  },
                  "city": "",
                  "cityLocation": Object {
                    "latitude": 0,
                    "longitude": 0,
                    "zoom": 0,
                  },
                  "coordinates": Array [
                    0,
                    0,
                  ],
                  "description": "",
                  "host": Object {
                    "avatar": "",
                    "isPro": true,
                    "name": "",
                  },
                  "id": 1,
                  "images": Array [],
                  "isFavorite": false,
                  "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                },
                Object {
                  "accommodation": Object {
                    "bedroomsCount": 0,
                    "features": Array [],
                    "guestsLimit": 0,
                    "isPremium": true,
                    "price": 241,
                    "rating": 4.4,
                    "title": "The Joshua Tree House",
                    "type": "room",
                  },
                  "city": "",
                  "cityLocation": Object {
                    "latitude": 0,
                    "longitude": 0,
                    "zoom": 0,
                  },
                  "coordinates": Array [
                    0,
                    0,
                  ],
                  "description": "",
                  "host": Object {
                    "avatar": "",
                    "isPro": true,
                    "name": "",
                  },
                  "id": 2,
                  "images": Array [],
                  "isFavorite": false,
                  "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                },
                Object {
                  "accommodation": Object {
                    "bedroomsCount": 0,
                    "features": Array [],
                    "guestsLimit": 0,
                    "isPremium": true,
                    "price": 241,
                    "rating": 4.4,
                    "title": "The Joshua Tree House",
                    "type": "room",
                  },
                  "city": "",
                  "cityLocation": Object {
                    "latitude": 0,
                    "longitude": 0,
                    "zoom": 0,
                  },
                  "coordinates": Array [
                    0,
                    0,
                  ],
                  "description": "",
                  "host": Object {
                    "avatar": "",
                    "isPro": true,
                    "name": "",
                  },
                  "id": 3,
                  "images": Array [],
                  "isFavorite": true,
                  "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                },
              ]
            }
            onFavoriteButtonClick={[Function]}
            typePage="main"
          />
        </section>
        <div
          className="cities__right-section"
        >
          <Map
            city="Amsterdam"
            typePage="main"
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`MainPage render correctly MainPage without offers 1`] = `
<div
  className="page page--gray page--main"
>
  <Header>
    <User
      onLinkEmailClick={[Function]}
    />
  </Header>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <CityList />
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container cities__places-container--empty container"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              No places to stay available
            </b>
            <p
              className="cities__status-description"
            >
              We could not find any property available at the moment in 
              Amsterdam
            </p>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          
        </div>
      </div>
    </div>
  </main>
</div>
`;
